src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/CaseDataDetailsGroupsRepository.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.digital.ho.hocs.casework.domain.repository;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.domain.model.CaseDataDetails;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static net.logstash.logback.argument.StructuredArguments.value;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.CASE_DATA_DETAILS_NOT_FOUND;
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.EVENT;
+
+@Service
+@Slf4j
+public class CaseDataDetailsGroupsRepository extends JsonConfigFolderReader {
+
+    public final Map<String, Map<String, List<CaseDataDetails.Fields>>> detailsFields;
+
+    public CaseDataDetailsGroupsRepository(ObjectMapper objectMapper) {
+        super(objectMapper);
+
+        detailsFields = readValueFromFolder(new TypeReference<CaseDataDetails>() {});
+    }
+
+    public Map<String, List<CaseDataDetails.Fields>> getGroupsDetailsFieldsByType(String type) {
```

Comment:
We likely could be smarter with this if we wanted to be with generics and more this to the base class. Unsure if its worth the hassle though.

Reply:
This can be for a second pass if we want to - I'm not convinced of the benefit here.